mmcv/ops/point_sample.py

```diff
@@ -181,25 +246,32 @@ def __init__(self, output_size, spatial_scale, aligned=True):
         self.aligned = aligned
 
     def forward(self, features, rois):
-
         num_imgs = features.size(0)
         num_rois = rois.size(0)
         rel_roi_points = generate_grid(
             num_rois, self.output_size, device=rois.device)
 
-        point_feats = []
-        for batch_ind in range(num_imgs):
-            # unravel batch dim
-            feat = features[batch_ind].unsqueeze(0)
-            inds = (rois[:, 0].long() == batch_ind)
-            if inds.any():
-                rel_img_points = rel_roi_point_to_rel_img_point(
-                    rois[inds], rel_roi_points[inds], feat.shape[2:],
-                    self.spatial_scale).unsqueeze(0)
-                point_feat = point_sample(
-                    feat, rel_img_points, align_corners=not self.aligned)
-                point_feat = point_feat.squeeze(0).transpose(0, 1)
-                point_feats.append(point_feat)
+        if torch.onnx.is_in_onnx_export():
+            feat = features[0].unsqueeze(0)
```

Comment:
Do we support exporting ONNX with batch here?

Reply:
No, it works only with batch size = 1. But as I understand, according to [this](https://github.com/open-mmlab/mmdetection/blob/master/tools/deployment/pytorch2onnx.py#L15) export script in `mmdetection`, batch > 1 is impossible here.

Reply:
Hi, currently, we are supporting exporting ONNX with batch and dynamic shape on [onnx branch](https://github.com/open-mmlab/mmdetection/tree/onnx) and we are planning to merge the onnx branch into master branch. It would be great if you could try to support `PointRend` model with batch and dynamic shape if possible.

Reply:
The original implementation converted with batch > 1. My bad. But in my case this fix is necessary. I added one more condition where is checked batch size. If it is possible I would like to remain code under `if torch.onnx.is_in_onnx_export() and num_imgs == 1`.

Reply:
Could you start with ONNX branch and support exporting ONNX with batch and dynamic shape if possible?

Reply:
Supporting Point Rend with batch and dynamic shape may need much work. We may do this together once the time is available.

Reply:
So, the code supports batch and dynamic shapes in onnx. [Here](https://github.com/DmitriySidnev/mmcv/blob/ds/grid_sample_test/grid_sample_test.py) is my test with `SimpleRoiAlign` class as an exporting model with the following dynamic axes: batch, height and width for input image and objects number for RoIs. But in my case, onnx model is an intermediate one and I need to avoid the loop through batch and get rid of indexing, it brakes my target graph in `OpenVINO`.

Reply:
Yes, once we support exporing to ONNX with batch, then there's no loop on batch dim in the ONNX file.

Reply:
In `OpenVINO` it leads to unsupported dynamic ops. Avoiding this loop fixes the result graph.

Reply:
Do you mean the exported PointRend ONNX file from the master branch would have `Loop` op in the graph?

Reply:
No, I mean that `OpenVINO` wrong processes indexing generated in loop. I do not know how exactly it works but removing the loop helps.